client/platform/desktop/backend/native/common.ts

```diff
@@ -0,0 +1,473 @@
+  }
+
+  const dsName = npath.parse(path).name;
+  const dsId = `${cleanString(dsName).substr(0, 20)}_${makeid(10)}`;
+
+  const jsonMeta: JsonMeta = {
+    version: JsonMetaCurrentVersion,
+    type: datasetType,
+    id: dsId,
+    fps: 5, // TODO
+    originalBasePath: path,
+    originalVideoFile: '',
+    originalImageFiles: [],
+    transcodedVideoFile: '', // TODO: this is empty (see above)
+    transcodedImageFiles: [], // TODO: this is empty
+    name: dsName,
+  };
+
+  /* TODO: Look for an EXISTING meta.json file to override the above */
+
+  if (datasetType === 'video') {
+    // get parent folder, since videos reference a file directly
+    jsonMeta.originalBasePath = npath.dirname(path);
+  }
+
+  const contents = await fs.readdir(jsonMeta.originalBasePath);
+
+  /* Extract and validate media from import path */
+  if (jsonMeta.type === 'video') {
+    jsonMeta.originalVideoFile = npath.basename(path);
+    const mimetype = mime.lookup(path);
+    if (mimetype) {
+      if (websafeImageTypes.includes(mimetype) || otherImageTypes.includes(mimetype)) {
+        throw new Error('User chose image file for video import option');
+      } else if (websafeVideoTypes.includes(mimetype)) {
+        /* TODO: Kick off video inspection and maybe transcode */
+      } else {
+        throw new Error(`unsupported MIME type for video ${mimetype}`);
+      }
+    } else {
+      throw new Error(`could not determine video MIME type for ${path}`);
+    }
+  } else if (datasetType === 'image-sequence') {
+    jsonMeta.originalImageFiles = contents.filter((filename) => {
+      const abspath = npath.join(jsonMeta.originalBasePath, filename);
+      const mimetype = mime.lookup(abspath);
+      /* TODO: support transcoding of non-web-safe image types */
+      return !!(mimetype && websafeImageTypes.includes(mimetype));
+    });
+    if (jsonMeta.originalImageFiles.length === 0) {
+      throw new Error(`no images found in ${path}`);
+    }
+  } else {
+    throw new Error('only video and image-sequence types are supported');
+  }
+
+  const projectDirAbsPath = await _initializeProjectDir(settings, jsonMeta);
+  await _saveAsJson(npath.join(projectDirAbsPath, JsonMetaFileName), jsonMeta);
+
+  let foundDetections = false;
+
+  /* Look for JSON track file as first priority */
+  const trackFileAbsPath = await _findJsonTrackFile(jsonMeta.originalBasePath);
+  if (trackFileAbsPath !== null) {
+    /* Move the track file into the new project directory */
+    await fs.move(
+      trackFileAbsPath,
+      npath.join(projectDirAbsPath, npath.basename(trackFileAbsPath)),
+    );
+    foundDetections = true;
+  }
+  /* Look for other types of annotation files as a second priority */
+  if (!foundDetections) {
+    const csvFileCandidates = contents
+      .filter((v) => CsvFileName.test(v))
+      .map((filename) => npath.join(jsonMeta.originalBasePath, filename));
+    const { fps, processedFiles } = await processOtherAnnotationFiles(
```

Comment:
very minor, in the JSON loading you warn the user if they have mutiple .json track files that are being loaded.  Here it will load multiple CSVs and only take one of them.  Reminder the processAnnotationFiles can take multiple CSVs because tracking pipelines will produce two csvs, a detector.csv and a tracking.csv.

Reply:
I thought about this when I was doing it, but IMO the current behavior is correct.

1) in normal operation, two result_.json files should never be created.  We wouldn't know which to load.
2) Likewise, multiple .csv files _can_ be created.  We still don't know which to load, so we allow the caller to specify a priority order because creation of these files is out of our control.

I don't know of a better strategy.  What would you suggest.

Reply:
I am talking about the caller being the media Importer here and that the mediaImporter should specify what occurs instead of leaving it up to filename order.  I wouldn't change how `processOtherAnnotationFiles`, but maybe here we handle in the return that there are multiple `processedFiles` and we either notify the user or error like we do with the JSON loader.  The specifics were importing an image sequence which had multiple csv files in the folder then it takes one of those CSV files (I think by filename order) and declares it as the trackJSON.  I was just including some background as to why `processOtherAnnotationFiles` will handle multiple csv files (mostly for my own recollection).